### PR TITLE
New Resource: azurerm_cosmosdb_postgresql_coordinator_configuration

### PR DESCRIPTION
**Pull request by neil-yechenwei**
image

### CODE REVIEW

File: internal/services/cosmos/cosmosdb_postgresql_coordinator_configuration_resource.go [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+				Properties: &configurations.ServerConfigurationProperties{},
+			}
+
+			if metadata.ResourceData.HasChange("value") {
+				parameters.Properties.Value = model.Value
+			}
+
+			if err := client.UpdateOnCoordinatorThenPoll(ctx, *id, *parameters); err != nil {
+				return fmt.Errorf("updating %s: %+v", *id, err)
+			}
```
Comment (catriona-m) [inline internal/services/cosmos/cosmosdb_postgresql_coordinator_configuration_resource.go:131]:
```suggestion
			if metadata.ResourceData.HasChange("value") {
				parameters := configurations.ServerConfiguration{
					Properties: &configurations.ServerConfigurationProperties{
						Value: model.Value,
					},
				}
				
				if err := client.UpdateOnCoordinatorThenPoll(ctx, *id, parameters); err != nil {
					return fmt.Errorf("updating %s: %+v", *id, err)
				}
			}
```

File: internal/services/cosmos/cosmosdb_postgresql_coordinator_configuration_resource.go [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+
+			state := CosmosDbPostgreSQLCoordinatorConfigurationModel{
+				Name:      id.CoordinatorConfigurationName,
+				ClusterId: configurations.NewServerGroupsv2ID(id.SubscriptionId, id.ResourceGroupName, id.ServerGroupsv2Name).ID(),
+			}
+
+			if props := model.Properties; props != nil {
+				state.Value = props.Value
+			}
```
Comment (catriona-m) [inline internal/services/cosmos/cosmosdb_postgresql_coordinator_configuration_resource.go:170]:
```suggestion
			state := CosmosDbPostgreSQLCoordinatorConfigurationModel{
				Name:      id.CoordinatorConfigurationName,
				ClusterId: configurations.NewServerGroupsv2ID(id.SubscriptionId, id.ResourceGroupName, id.ServerGroupsv2Name).ID(),
			}

			if model := resp.Model; model != nil {
				if props := model.Properties; props != nil {
					state.Value = props.Value
				}
			}
```

File: internal/services/cosmos/cosmosdb_postgresql_coordinator_configuration_resource.go [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+
+			parameters := &configurations.ServerConfiguration{
+				Properties: &configurations.ServerConfigurationProperties{
+					Value: defaultValue,
+				},
+			}
+
+			if err = client.UpdateOnCoordinatorThenPoll(ctx, *id, *parameters); err != nil {
+				return fmt.Errorf("deleting %s: %+v", id, err)
+			}
```
Comment (catriona-m) [inline internal/services/cosmos/cosmosdb_postgresql_coordinator_configuration_resource.go:209]:
```suggestion
			defaultValue := ""
			if resp.Model != nil && resp.Model.Properties != nil && resp.Model.Properties.DefaultValue != nil {
				defaultValue = *resp.Model.Properties.DefaultValue
			}

			parameters := configurations.ServerConfiguration{
				Properties: &configurations.ServerConfigurationProperties{
					Value: defaultValue,
				},
			}

			if err = client.UpdateOnCoordinatorThenPoll(ctx, *id, parameters); err != nil {
				return fmt.Errorf("deleting %s: %+v", *id, err)
			}
```

**Review by catriona-m — CHANGES_REQUESTED** [review]
Thanks for this @neil-yechenwei, I left a couple of comments inline for you to consider but otherwise this is looking pretty good so far.

File: internal/services/cosmos/cosmosdb_postgresql_coordinator_configuration_resource.go (hunk 3 above)
Comment (neil-yechenwei) [reply internal/services/cosmos/cosmosdb_postgresql_coordinator_configuration_resource.go]: Updated

File: internal/services/cosmos/cosmosdb_postgresql_coordinator_configuration_resource.go (hunk 2 above)
Comment (neil-yechenwei) [reply internal/services/cosmos/cosmosdb_postgresql_coordinator_configuration_resource.go]: Updated

File: internal/services/cosmos/cosmosdb_postgresql_coordinator_configuration_resource.go (hunk 1 above)
Comment (neil-yechenwei) [reply internal/services/cosmos/cosmosdb_postgresql_coordinator_configuration_resource.go]: Updated

File: internal/services/cosmos/cosmosdb_postgresql_coordinator_configuration_resource.go [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+
+			parameters := &configurations.ServerConfiguration{
+				Properties: &configurations.ServerConfigurationProperties{
+					Value: model.Value,
+				},
+			}
+
+			if err := client.UpdateOnCoordinatorThenPoll(ctx, id, *parameters); err != nil {
+				return fmt.Errorf("updating %s: %+v", id, err)
+			}
```
[the remark this answers is not in the document]
Comment (neil-yechenwei) [reply internal/services/cosmos/cosmosdb_postgresql_coordinator_configuration_resource.go]: Updated

File: internal/services/cosmos/cosmosdb_postgresql_cluster_resource.go [hunk 5]
```diff
@@ -16,6 +16,8 @@ import (
 	"github.com/hashicorp/terraform-provider-azurerm/utils"
 )
 
+var CosmosDbPostgreSQLClusterResourceName = "azurerm_cosmosdb_postgresql_cluster"
```
[the remark this answers is not in the document]
Comment (neil-yechenwei) [reply internal/services/cosmos/cosmosdb_postgresql_cluster_resource.go:19]: Updated

**Comment by neil-yechenwei** [comment]
@catriona-m , thanks for the comments. I updated PR. Please take another look. Thanks.

**Review by catriona-m — APPROVED** [review]
Thanks @neil-yechenwei - it looks like merging another azurerm_cosmosdb_postgresql* has caused conflicts with this and the other azurerm_cosmosdb_postgresql* PRs, but otherwise this LGTM!

**Pull request merged**